atlite/gis.py

```diff
@@ -459,15 +458,15 @@ def shape_availability(geometry, excluder):
             masked_ = ~(masked_).astype(bool)
         if d["buffer"]:
             iterations = int(d["buffer"] / excluder.res) + 1
-            masked_ = dilation(masked_, iterations=iterations).astype(int)
+            masked_ = dilation(masked_, iterations=iterations)
 
-        exclusions.append(masked_.astype(int))
+        exclusions = exclusions | masked_
 
     for d in excluder.geometries:
         masked = ~geometry_mask(d["geometry"], shape, transform, invert=d["invert"])
-        exclusions.append(masked.astype(int))
+        exclusions = exclusions | masked
 
-    return (sum(exclusions) == 0).astype(float), transform
+    return ~exclusions, transform
```

Comment:
```suggestion
    return (~exclusions).astype(float), transform
```

Else we are changing the `dtype` of the returned `numpy.array` and break e.g. the examples in the documentation and probably a lot of workflows... .

Reply:
(It broke some of my workflows)

Reply:
It also doesn't solve the issue of changing datatype. The returned `dtype` is still a boolean, not a float.

Reply:
Testing the memory profiling and whether it works would be nice. 
The effective change now is:
* The returned mask out of shape_availibility is of dtype boolean. We do not need a float or int here, since it is simply given the information whether an area is eligible or not. The reason why the return type was a float becomes clear in the next point.
* A conversion to `uint8` in the rasterio `reproject` function in `shape_availability_reprojected` was added, converting the boolean mask coming out of `shape_availibility`. The `reproject` function does not like booleans, as it takes the mean out of multiple cells and combines them into a float representing the value of the bigger/repojected cell. However having a integer input dtype seems to be totally feasible and reduces memory usage.

Reply:
Ok, I'll run the tests again.

1. Regarding the return type: We technically don't need a float here, but we are changing the returned datatype of a function. This is a **breaking change** which I think we should avoid. We have some examples which also break due to this change. Returning an int or keeping the return type as float doesn't hurt much (shouldn't increase memory consumption significantly) but avoids the breaking change

2. If I understand you correctly 1. would also make the changes in 2. obsolete?